### PR TITLE
Add Low speed connector support for 96Boards IE boards

### LINKED ISSUE #15598: Standard devicetree connectors for boards
**Is your feature request related to a problem? Please describe.**
For the boards belonging to a standard form factor like 96Boards, it would be nice to have a dedicated devicetree connector which can expose the available peripherals like GPIO, I2C, SPI in a generic way so that the sample applications can run on any board adhering to the standard.

**Describe the solution you'd like**
This could be achieved by having a below devicetree node in board dts:

```
        96b-ls-con {
                compatible = "96b-ls-con";
                i2c0 = &i2c0;
                uart0 = &uart0;

                gpioa: gpioa {
                        gpios = <&gpio0 1 0>;
                        label = "GPIO-A";
                };
                gpiob: gpiob {
                        gpios = <&gpio0 2 0>;
                        label = "GPIO-B";
                };
        };
```

Then the applications can make use of the definitions like `DT_96B_LS_CON_GPIOA_GPIO_CONTROLLER`, `DT_96B_LS_CON_GPIOA_GPIO_PIN` etc... 

For achieving the above solution, a separate connector binding needs to be created and it shall include all possible peripherals.

**Describe alternatives you've considered**
There is already a concept of shields exist in Zephyr but it is limited to the expansion connectors and specifically not able to handle individual gpio's.

### PR DESCRIPTION
**Pull request by Mani-Sadhasivam**
All 96Boards complying to the IE spec exposes either 40pin or 30pin
standard low speed connectors for peripheral connectivity. These
connectors are well defined and available in the IE spec [1]. This PR
creates two different gpio bindings for these connectors:

1. 96b-lscon-1v8 - 40 pin connector with maximum of 12 GPIOs
2. 96b-lscon-3v3 - 30 pin connector with maximum of 8 GPIOs

These bindings will be utilized by the 96Boards for exposing the
GPIO pins as nexus node as per the devicetree spec. This will allow
the shields and applications to use board independent GPIO mapping.

While at it, 96Boards generic node labels for I2C, SPI and UART are also
added.

[1] https://linaro.co/ie-specification

Fixes: #15598

Signed-off-by: Manivannan Sadhasivam <mani@kernel.org>

### CODE REVIEW

File: dts/bindings/gpio/96boards-lscon-3v3.yaml
Comment (erwango) [inline dts/bindings/gpio/96boards-lscon-3v3.yaml]: I though using numbers as first character was prohibited.
Isn't it still the case ?

**Review by erwango — CHANGES_REQUESTED** [review]
Please update boards documentation so we can quickly reach connector specification.
Otherwise, change is good.
Little hick up is that we miss sample to test that. Maybe we can add boards to `tests/drivers/gpio/gpio_basi_api` once topic-gpio branch is merged.

**Comment by Mani-Sadhasivam** [comment]
@erwango Okay, will add the link to 96Boards IE spec.

> Little hick up is that we miss sample to test that. Maybe we can add boards to tests/drivers/gpio/gpio_basi_api once topic-gpio branch is merged.

Sure. I can add that later.

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/96b_carbon/96b_lscon.dtsi
Comment (Mani-Sadhasivam) [reply boards/arm/96b_carbon/96b_lscon.dtsi]: Done

File: boards/arm/96b_wistrio/doc/96b_wistrio.rst
Comment (dbkinder) [inline boards/arm/96b_wistrio/doc/96b_wistrio.rst]: sorry for the nit change, but change ``platform`` to ``platforms``

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by galak** [comment]
Any updates on this?

**Comment by Mani-Sadhasivam** [comment]
@galak AFIAK, this PR is close to get merged. Just waiting for the approvals from @dbkinder and @erwango

**Review by dbkinder — APPROVED** [review]
+1 for doc changes, thanks.

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**
